sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -138,7 +138,10 @@ std::vector<EventImplPtr> Scheduler::getWaitList(EventImplPtr Event) {
 
 void Scheduler::waitForEvent(EventImplPtr Event) {
   std::shared_lock<std::shared_timed_mutex> Lock(MGraphLock);
-  GraphProcessor::waitForEvent(std::move(Event));
+  // The command might have been cleaned up (and set to nullptr) by another
+  // thread
+  if (Event->getCommand())
```

Comment:
One more thing. I think, we should move this check inside `GraphProcessor:;waitForEvent` to make it safe.